src/app/services/reducers/permissions-reducer.ts

```diff
@@ -55,3 +56,35 @@ export function scopes(state: IScopes = initialState, action: IAction): any {
       return state;
   }
 }
+
+const unconsentInitialState: IUnconsent = {
+  pending: false,
+  error: null,
+  data: []
+}
+
+export function unconsentingScopes(state: IUnconsent = unconsentInitialState, action: IAction): any {
+  switch (action.type) {
+    case REVOKE_PERMISSION_PENDING:
+      return {
+        pending: action.response,
+        error: null,
+        data: []
+      }
+    case REVOKE_PERMISSION_SUCCESS:
+      return {
+        pending: false,
+        error: null,
+        data: action.response
+      }
+
+    case REVOKE_PERMISSION_ERROR:
+      return {
+        pending: false,
+        error: action.response,
+        data: []
+      }
+    default:
+      return state;
+  }
+}
```

Comment:
Since we do not store the permissions that we have unconsented to, why not reuse the consenting state objects?

Reply:
Yes. We will re-use those